File: baekjoon/src/queue/No10866.java
package queue;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Deque;
import java.util.LinkedList;

/**
 * https://www.acmicpc.net/problem/10866
 */
public class No10866 {
    public static Deque<Integer> deque = new LinkedList<>();

    public static StringBuilder builder = new StringBuilder();

    public static void command(BufferedReader reader) throws IOException {
        String[] orders = reader.readLine().split(" ");
        if (orders.length == 2) {
            switch (orders[0]) {
                case "push_back":
                    deque.addLast(Integer.parseInt(orders[1]));
                    break;
                case "push_front":
                    deque.addFirst(Integer.parseInt(orders[1]));
                    break;
            }
        } else {
            switch (orders[0]) {
                case "pop_front":
                    builder.append(deque.isEmpty() ? -1 : deque.pollFirst()).append("\n");
                    break;
                case "pop_back":
                    builder.append(deque.isEmpty() ? -1 : deque.pollLast()).append("\n");
                    break;
                case "size":
                    builder.append(deque.size()).append("\n");
                    break;
                case "empty":
                    builder.append(deque.isEmpty() ? 1 : 0).append("\n");
                    break;
                case "front":
                    builder.append(deque.isEmpty() ? -1 : deque.peekFirst()).append("\n");
                    break;
                case "back":
                    builder.append(deque.isEmpty() ? -1 : deque.peekLast()).append("\n");
                    break;
            }
        }
    }

    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        int numberOfInput = Integer.parseInt(reader.readLine());
        for (int i = 0; i < numberOfInput; i++)
            command(reader);
        System.out.println(builder);
    }
}

File: baekjoon/src/tags/data_structures/No10866.kt
package tags.data_structures

import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import java.util.LinkedList

/**
 * https://www.acmicpc.net/problem/10866
 */
fun main() {
    val reader = BufferedReader(InputStreamReader(System.`in`))
    val writer = BufferedWriter(OutputStreamWriter(System.out))
    val deque = LinkedList<Int>()

    val numberOfInput = reader.readLine().toInt()
    repeat(numberOfInput) {
        val operations = reader.readLine().split(" ")
        if (operations[0] == "push_front")
            deque.addFirst(operations[1].toInt())
        else if (operations[0] == "push_back")
            deque.addLast(operations[1].toInt())
        else if (operations[0] == "pop_front") {
            writer.write("${if (deque.isEmpty()) -1 else deque.pollFirst()}")
            writer.newLine()
        } else if (operations[0] == "pop_back") {
            writer.write("${if (deque.isEmpty()) -1 else deque.pollLast()}")
            writer.newLine()
        } else if (operations[0] == "size") {
            writer.write("${deque.size}")
            writer.newLine()
        } else if (operations[0] == "empty") {
            writer.write("${if (deque.isEmpty()) 1 else 0}")
            writer.newLine()
        } else if (operations[0] == "front") {
            writer.write("${if (deque.isEmpty()) -1 else deque.peekFirst()}")
            writer.newLine()
        } else if (operations[0] == "back") {
            writer.write("${if (deque.isEmpty()) -1 else deque.peekLast()}")
            writer.newLine()
        }
    }
    writer.flush()
}
